homeassistant/helpers/reload.py

```diff
@@ -77,6 +77,19 @@ async def _resetup_platform(
 
         root_config[integration_platform].append(p_config)
 
+    # If new adr0007 style, include that as well.
+    if conf.get(integration_name):
+        # Check if it's a multi-platform config
+        added = False
+        for item in conf[integration_name]:
+            if isinstance(item, dict):
+                if item.get(integration_platform):  # Config for current platform
```

Comment:
```suggestion
            if isinstance(item, dict) and (current_platform_config := item.get(integration_platform)):
```

Reply:
Resolved in https://github.com/home-assistant/core/commit/41795a16be20f077c4b551daeee664a3c3342ba9